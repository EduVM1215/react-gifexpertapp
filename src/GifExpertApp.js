import React, { useState } from "react";
import { AddCategory } from "./Components/AddCategory";
import { GifGrid } from "./Components/GifGrid";

export const GifExpertApp = ({defaultCategories = []}) => {
  //mandamos el setCategories como prop al component
  const [categories, setCategories] = useState(defaultCategories);

  /*const handleAgregar = ()=>{

      setCategories([...categories, 'Jordi Wild']);
      //setCategories(['Jordi Wild', ...categories]);
      // setCategories (cats =>[...cats, 'Jordi Wild']}

    }*/

  return (
    <>
      <h2>GifExpertApp</h2>
      <AddCategory setCategories={setCategories} />
      <hr />

      <ul>
        {
          categories.map((category) => (
            <GifGrid 
              key={category} 
              category={category} />
          ))
        }
      </ul>
    </>
  );
};
