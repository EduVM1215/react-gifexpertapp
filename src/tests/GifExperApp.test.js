import React from 'react';
import '@testing-library/jest-dom'
import {shallow} from 'enzyme'
import {GifExpertApp} from '../GifExpertApp'


describe('pruebas sobre GifExpertApp', () => { 

    const wrapper=shallow(<GifExpertApp />)

    test('Debe mostrar el componente correctamente', ()=>{

    
        expect(wrapper).toMatchSnapshot();
    })


    test('debe de mostrar una lista de categorias', ()=>{
        
        const categories = ['Paco', 'Pepe']
        const wrapper=shallow(<GifExpertApp defaultCategories={categories}/>)

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length)
    })
 })