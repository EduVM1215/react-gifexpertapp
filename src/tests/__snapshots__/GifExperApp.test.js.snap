// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas sobre GifExpertApp Debe mostrar el componente correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ul />
</Fragment>
`;

exports[`pruebas sobre GifExpertApp debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ul>
    <GifGrid
      category="Paco"
      key="Paco"
    />
    <GifGrid
      category="Pepe"
      key="Pepe"
    />
  </ul>
</Fragment>
`;
